### ISSUE TITLE
CRIU package for Ubuntu 24.04

### ISSUE THREAD

**Issue by yang** [checkpoint-restore/criu#2404]
I saw that criu was removed from the Ubuntu 24's apt repos - is there another easy package based way to install on Ubuntu, or is build chain the only way from here on? Thanks.

https://launchpad.net/ubuntu/noble/amd64/criu/3.17.1-3

https://launchpad.net/~criu/+archive/ubuntu/ppa (no noble releases)

**Comment by mihalicyn** [comment]
It's not removed from official Ubuntu repos, we just don't have PPA builds for Noble yet.

**Comment by mihalicyn** [comment]
Related to https://github.com/checkpoint-restore/criu/issues/2398

**Comment by rst0git** [comment]
I will update the OBS/PPA packages with the workaround from https://github.com/checkpoint-restore/criu/issues/2398#issuecomment-2079198028 for the time being and enable support for Ubuntu 24.04.

**Comment by mihalicyn** [comment]
Huge thanks, Radostin!

I'll try to get CRIU into the official repos too. I was able to workaround compile-time failure, but there is another problem related to crit and python (see https://bugs.launchpad.net/ubuntu/+source/criu/+bug/2066148).

**Comment by rst0git** [comment]
>  Please avoid running ``setup.py`` and ``easy_install``.

Should be fixed by https://github.com/checkpoint-restore/criu/pull/2282

In addition, have also applied the following patch in https://github.com/rst0git/criu-deb-packages/commit/80ddb47974f541b99a7afc3e4e226e850648f27d

**Comment by rst0git** [comment]
@yang I can confirm that Ubuntu 24.04 support has been enabled for the packages in OBS and PPA:
- https://build.opensuse.org/package/show/devel:tools:criu/criu
- https://launchpad.net/~criu/+archive/ubuntu/ppa

**Comment by mihalicyn** [comment]
As I can see from https://launchpad.net/~criu/+archive/ubuntu/ppa/+packages we have no builds for Noble, but only for Jammy. I guess we need to change the last entry in `debian/changelog` file and specify `noble` in there.

I'm currently stuggling with this install-stage failure during binary package build:
https://launchpadlibrarian.net/730947573/buildlog_ubuntu-noble-amd64.criu_3.17.1-3ubuntu1_BUILDING.txt.gz

I have applied these patches:
https://github.com/checkpoint-restore/criu/commit/7f0f07599a680258092a7790fb3046856f486f2f
https://github.com/rst0git/criu-deb-packages/commit/80ddb47974f541b99a7afc3e4e226e850648f27d

but it looks like we need something in addition to make it work on Noble. Will continue investigation.

**Comment by rst0git** [comment]
@mihalicyn Thanks for pointing this out! I've updated the release to `noble` in [launchpad](https://launchpad.net/~criu/+archive/ubuntu/ppa/+packages).

```
dh_usrlocal: error: debian/criu/usr/local/bin/crit is not a directory
make: *** [debian/rules:9: binary] Error 25
```
There is a bug in the version of pip available since Ubuntu 22.04. The `pip install` command does not take into account the value specified with `--prefix` when installing scripts: This problem does not exist in earlier versions and it has fixed in newer versions of pip. We use the following workaround for this problem: https://github.com/rst0git/criu-deb-packages/commit/e025b7edc7a1d6454048e337f774748de846cdf0

**Comment by mihalicyn** [comment]
Thanks @rst0git!

I have prepared a new debdiff:
https://bugs.launchpad.net/ubuntu/+source/criu/+bug/2066148/comments/5

hope it will be sponsored and uploaded.

**Comment by avagin** [comment]
@rst0git do we have our repo with the up-to-date version?

**Comment by rst0git** [comment]
@Brandl you can install the latest version of CRIU on Ubuntu 24.04 from [Launchpad](https://launchpad.net/~criu/+archive/ubuntu/ppa) or [OpenBuildService](https://build.opensuse.org/package/show/devel:tools:criu/criu).

**Comment by Cwavs** [comment]
Hi, Just wondering if there's any updates on the package in the official repos? I've hit a roadblock as it appears that the version of crun built in 24.04.2 lacks checkpoint support and I'm not sure if this could be related to why. I'm gonna probably install a fresh build of crun myself to hopefully fix this, but just wanted to check in on the issue seen as it's been a while since the last update.

**Comment by rst0git** [comment]
Closing this issue as the original problem has been resolved.

**Comment by Cwavs** [comment]
> > I've hit a roadblock as it appears that the version of crun built in 24.04.2 lacks checkpoint support and I'm not sure if this could be related to why. I'm gonna probably install a fresh build of crun myself to hopefully fix this
> 
> You can also replace `crun` with `runc`. It should work out of the box.

@rst0git Yeah, I could, I was just hoping to not as it means I have to install runc on the new machine to restore, and I have to restart all running containers to get them to move over to runc. Thanks for the help though.
